nyaml/nyaml2nxdl.py

```diff
@@ -476,13 +479,15 @@ def xml_handle_dim(obj, value):
                     valid_dims.append(entry)
             if len(valid_dims) > 0:
                 dims = ET.SubElement(obj, "dimensions")
-                dims.set("rank", str(len(valid_dims)))
+                # dims.set("rank", str(len(valid_dims)))
```

Comment:
So that, `nxdl.xml` and `yaml` are reproducible which is the another main motivation of this tool. That why you see `group_name(NXgroup)` and `(NXgroup)group_name` are important. So that the `name` and `group` in the definition does not lose the order though still the element has the same meaning with varied order.

Reply:
It would only change the nyaml files which are anyway only in FAIRmat so far. But fine for me to keep it as it is.